dial_sync.go

```diff
@@ -53,22 +54,38 @@ func (ad *activeDial) incref() {
 
 func (ad *activeDial) decref() {
 	ad.refCntLk.Lock()
-	defer ad.refCntLk.Unlock()
 	ad.refCnt--
```

Comment:
Since you are no longer using defer anywhere in this function I am going to assume that nothing can panic otherwise the locks might never be released.